controller/node_controller.go

```diff
@@ -313,16 +321,17 @@ func (nc *NodeController) syncNode(key string) (err error) {
 			for _, podCondition := range podConditions {
 				if podCondition.Type == v1.PodReady {
 					if podCondition.Status == v1.ConditionTrue && pod.Status.Phase == v1.PodRunning {
-						node.Status.Conditions = types.SetConditionAndRecord(node.Status.Conditions,
-							longhorn.NodeConditionTypeReady, longhorn.ConditionStatusTrue,
-							"", fmt.Sprintf("Node %v is ready", node.Name),
-							nc.eventRecorder, node, v1.EventTypeNormal)
+						// create the event after `UpdateNodeStatus` without the error
+						nodeIsReadyEventMsg = fmt.Sprintf("Node %v is ready", node.Name)
+						nodeIsReadyEventType = v1.EventTypeNormal
```

Comment:
NIT: You can set Msg and Type only when there is a change. Then you don't need `nodeIsReadyEventChanged`

Reply:
Modified. Removed `nodeIsReadyEventChanged`. Thanks.